### PR TITLE
Fix eslint errors

### PR DESCRIPTION
**Pull request by marcin-gromek-wttech**
Done:

*   Added Eslint and reformatted all files based on eslint rules
*   Rewrite `BurgerBuilder.js` to use "current" conventions/practices of writing React components
*   add props validation via `PropTypes`
*   move costans props to `utils` directory

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,20 +1,23 @@
...
 function App() {
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+
   return (
     <Layout>
       <Switch>
-        <Route exact path='/' component={BurgerBuilder} />
-        <Route path='/checkout' component={Checkout} />
-        <Route path='/orders' component={Orders} />
+        <AuthContext.Provider value={{ isLoggedIn: isLoggedIn }}>
```
Comment (marcin-gromek-wttech) [inline src/App.js:16]: Added provider for React Context - will continuing to work on login page as a next step - i will need it

File: src/App.js [hunk 2]
```diff
@@ -1,20 +1,23 @@
...
+import { Route, Switch } from 'react-router-dom';
+import React, { useState } from 'react';
+import AuthContext from './store/auth-context';
+import BurgerBuilder from './containers/BurgerBuilder/BurgerBuilder';
+import Checkout from './containers/Checkout/Checkout';
+import Layout from './hoc/Layout/Layout';
+import Orders from './containers/Orders/Orders';
 
 function App() {
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
```
Comment (marcin-gromek-wttech) [inline src/App.js:11]: Will continuing to work on login page as a next step - i will need it

File: src/components/Login/Login.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
```
Comment (marcin-gromek-wttech) [inline src/components/Login/Login.js:1]: Will continuing to work on login page as a next step - i will need it

File: src/hoc/withErrorHandler/withErrorHandler.js [hunk 4]
```diff
@@ -1,44 +1,49 @@
...
-      axios.interceptors.response.eject(this.responseInterceptor);
-    }
-
-    errorConfirmedHandler = () => {
-      this.setState({ error: null });
+import React, { Component } from 'react';
+import Modal from '../../components/Modal/Modal';
+import Aux from '../Aux/Aux';
+
+const withErrorHandler = (WrappedComponent, axios) => class extends Component {
```
Comment (marcin-gromek-wttech) [inline src/hoc/withErrorHandler/withErrorHandler.js:5]: Error Boundaries are only available in Class-based React Components. There [is currently not a way to implement it using React Hooks](https://reactjs.org/docs/hooks-faq.html#do-hooks-cover-all-use-cases-for-classes).

**Review by piotr-michalczuk-wttech — APPROVED** [review]

File: src/App.js (hunk 1 above)
Comment (marcin-gromek-wttech) [reply src/App.js:16]: I replaced React Context with React Redux in PR https://github.com/marcin-gromek-wttech/burger-builder/pull/2 
...also implemented re-ducks approach

File: src/App.js (hunk 2 above)
Comment (marcin-gromek-wttech) [reply src/App.js:11]: I replaced React Context with React Redux in PR https://github.com/marcin-gromek-wttech/burger-builder/pull/2 ...also implemented re-ducks approach

**Pull request merged**